src/sentry/grouping/api.py

```diff
@@ -100,15 +101,27 @@ def _get_enhancements(self, project: Project) -> str:
         cache_prefix = self.cache_prefix
         cache_prefix += f"{LATEST_VERSION}:"
         cache_key = (
-            cache_prefix + md5_text(f"{enhancements_base}|{project_enhancements}").hexdigest()
+            cache_prefix
+            + md5_text(
+                f"{enhancements_base}|{automatic_enhancements}|{project_enhancements}"
+            ).hexdigest()
         )
         enhancements = cache.get(cache_key)
         if enhancements is not None:
             return enhancements
 
         try:
+            # Automatic enhancements are always applied first, so they take precedence over
+            # project-specific enhancements.
```

Comment:
I think you have the description of the precedence backwards.

```suggestion
            # Automatic enhancements are always applied first, so they can be overridden by
            # project-specific enhancements.
```